drivers/saul/init_devs/auto_init_saul_nrf_vddh.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     sys_auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization of internal voltage sensor directly mapped to SAUL reg
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "log.h"
+#include "saul_reg.h"
+#include "saul/periph.h"
+#include "periph/adc.h"
+
+/* ADC uses ¼ Vdd for Vref amplified by 4 */
+#define VDD_REF_mV  (3300)
```

Comment:
This seems potentially broken: when using VDDH, VDD is not fixed to 3.3V, but it depends on the value of `UICR->REGOUT0` and can have a value from 1V8 to 3V3... So I guess the correct value should be read from that register.

Reply:
fixed